music21/musicxml/m21ToXml.py

```diff
@@ -3472,7 +3521,23 @@ def noteToXml(self, n: note.GeneralNote, noteIndexInChord=0, chordParent=None):
             for t in mxTieList:
                 mxNote.append(t)
 
-        # TODO: instrument
+        # instrument tags are necessary when there is more than one
+        if self.parent is not None and len(self.parent.instrumentStream) > 1 and not n.isRest:
+            if chordParent:
```

Comment:
Maybe:  

```
searchingObject = chordParent if chordParent else n
closest_inst = searchingObject.getContextByClass(Instrument, followDerivation=False)
```

to remove the redundancy.

Reply:
Same implementation as getInstrument(), but with the ability to store a different instrument, so I was planning on using that now that #1033 is in:

https://github.com/cuthbertLab/music21/blob/fe2586ad956fa68e048b979bba2d15eb6369bf9c/music21/note.py#L1205-L1211

Reply:
I see the point that it needs to be tested first.

Reply:
AH, you already pushed that change. Got it. Just needs a doctest.